app/models/order.rb

```diff
@@ -9,22 +9,36 @@ class Order < ApplicationRecord
   accepts_nested_attributes_for :order_items, allow_destroy: true
 
   before_create do
-    product_tax = TAX
-    user.cart.cart_items.each do |cart_item|
-      order_items.build(product_id: cart_item.product_id, quantity: cart_item.quantity, price_with_tax: cart_item.price_with_tax * (1 + product_tax))
-    end
+    self.product_tax = TAX
+    set_order_items
     user.cart.destroy!
   end
 
+  after_create do
+    set_postage
```

Comment:
ここもかな。
```suggestion
    update_postage
```

after_create で実行している理由ってなぜ？（before_createでも個数は求められそうに見えたので）